rfc3164/rfc3164.go

```diff
@@ -285,6 +286,22 @@ func fixTimestampIfNeeded(ts *time.Time) {
 
   newTs := time.Date(y, ts.Month(), ts.Day(), ts.Hour(), ts.Minute(),
     ts.Second(), ts.Nanosecond(), ts.Location())
+  lastYearTs := time.Date(y-1, ts.Month(), ts.Day(), ts.Hour(), ts.Minute(),
+    ts.Second(), ts.Nanosecond(), ts.Location())
+  nextYearTs := time.Date(y+1, ts.Month(), ts.Day(), ts.Hour(), ts.Minute(),
+    ts.Second(), ts.Nanosecond(), ts.Location())
+
 
-  *ts = newTs
+  /* When guessing the year of an event, pick the date that's closer to the current date 
+     - the correct date may be in the future */
```

Comment:
I'm having a little trouble following this.  I'm open to the idea that this is just a complex calculation and it's not possible to make the code clearer.  If that's the case can we have a more thorough explination in the code comment?